figure/figure.go

```diff
@@ -36,6 +36,27 @@ type Rhombus struct {
 	text                string
 }
 
+type Input struct {
+	x, y, width, height int
+	text                string
+}
+
+func (i *Input) draw(canvas *gg.Context) {
+	x := i.x - i.width/2 + horizontalMargins/2
+	y := i.height + i.y
+	canvas.MoveTo(float64(x), float64(i.y))
+	x2 := i.x + i.width/2
+	canvas.LineTo(float64(x2), float64(i.y))
+	x3 := x2 - horizontalMargins/2
+	canvas.LineTo(float64(x3), float64(y))
+	x4 := x - horizontalMargins/2
+	canvas.LineTo(float64(x4), float64(y))
+	canvas.LineTo(float64(x), float64(i.y))
+	canvas.Stroke()
+	middleY := i.height/2 + i.y
+	canvas.DrawStringAnchored(i.text, float64(i.x), float64(middleY), 0.5, 0.35)
```

Comment:
The names like `x2, x3, x4`, etc. are a bit confusing. Maybe it's better to use long names like `bottom_left`, `top_right`, or at least have comments what each step does?

And as I understand, the `horizontalMargins` controls angles of the parallelogram, doesn't it?